### ISSUE TITLE
[Bug]: query/data/index's recovery logic should handle dropping collections due to asynchronous recovery of rootcoord

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#21425]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Consider this case:

- 1, rootcoord received `drop` request;
- 2, rootcoord marked the collection as `dropping` and then do the GC asynchronous;

GC includes:
- a, releasing collection in querynodes;
- b, dropping all indexes;
- c, notifying datanodes to gc collection;

If we deployed standalone Milvus, and Milvus crashed before `a` step is done, according to the recovery design, querycoord will think the collection should be reloaded but the rootcoord think the collection is dropping and will do the GC continuely and asynchronously. Both querycoord and rootcoord are right in their own perspectives, rootcoord is under normal state (available to other components) when it's doing recovery asynchronously, so other components which depend on rootcoord can connect to rootcoord and tried to do recovery, for example querycoord tried to reload these dropping collections and they will get "Collection Not Exists" exception and then exit themselves. Under this situation, rootcoord has no chance to complete the GC, neither, and thus will cause Milvus can't be restarted forever.

One simple solution is to let other components handle "Collection Not Exists" gracefully, this exception should be ignored and we just need to print a warning. For now, I can't image other cases will lead other components get this exception.

By the way, we can't make rootcoord's recovery logic to be synchronous. For example, if we make rootcoord available after all GC is complete, this will lead to a cycle dependence, rootcoord depends on other components to complete recovery and then to be available, other components depend on rootcoord to continue the service, so they both can't go to be healthy.

**Comment by longjiquan** [comment]
This is the root cause of #21409

**Comment by xiaofan-luan** [comment]
can we put step a,b,c before drop the collection?
at least for release/index, it should be done before collection is dropped
That might the step more smooth.

**Comment by longjiquan** [comment]
@xiaofan-luan I'm afraid we can't do this, since you cannot tell the users collection is not dropped after you released the collection or dropped the indexes.

**Comment by longjiquan** [comment]
Fixed in #21471 and #21628
